### ISSUE TITLE
Inconsistent error in background subtraction

### ISSUE THREAD

**Issue by msmcfarlin** [vladan-stojnic/Detection-of-Flying-Honeybees-in-UAV-Videos#9]
Hello,

I am running through your tutorial and just generated 1000 videos with synthetic bees. I am now trying to run a for loop with the background subtraction function below...

```
for i in ./*.MP4;
do 
python /home/user/Detection-of-Small-Flying-Objects/bgsub.py -i $i -o /home/user/bee_workspace/background_sub/$i --num_avg 50
done
```

The function runs successfully on 660 of the files and generates the expected background subtraction output file. On the remaining 340 I get the error below and no output video.

            Traceback (most recent call last):
              File "/home/mmcfarlin/Detection-of-Small-Flying-Objects/bgsub.py", line 65, in <module>
                background_subtraction_loop(infile, bgsub_out, num_avg)
              File "/home/mmcfarlin/Detection-of-Small-Flying-Objects/bgsub.py", line 55, in background_subtraction_loop
                writer_bgsub.close()
              File "/home/mmcfarlin/.conda/envs/bee4exp/lib/python3.6/site-packages/skvideo/io/ffmpeg.py", line 430, in close
                if self._proc is None:  # pragma: no cover
            AttributeError: 'FFmpegWriter' object has no attribute '_proc'

I get the same result if I try running the background subtraction function out of the for loop on one of the synthetic bee videos that failed. I'm not sure why this is happening on only a third of the "bee added" videos or how to resolve it. Have you ever encountered this issue?

Thank you so much for your time.
-Mike

**Comment by vladan-stojnic** [comment]
Hi,

We didn't have that problem. Is it happening only on one video (661) or on all of the rest 340? Also I think FFMpeg can sometimes mask real issues in the code by its stack trace (see [1](https://stackoverflow.com/questions/47557983/attributeerror-ffmpegwriter-object-has-no-attribute-proc), [2](https://github.com/scikit-video/scikit-video/issues/139)). So maybe try putting `writer_bgsub._proc = None` somewhere right after creating the writer, this should help you find if some other line is actually throwing and error.

**Comment by msmcfarlin** [comment]
The error is happening on all of the 340 videos, and they don't appear to be in any order. For example, in the first ten bee added videos, videos 2, 4, 5, and 9 produce the error. I'll try adding `writer_bgsub._proc = None` and let you know what happens.

Thank you.

**Comment by msmcfarlin** [comment]
Apologies for the late response, I got distracted after the holidays.

Regarding this issue, I went back through the initial steps in the protocol and realized that when I ran the stabilization step that it did not complete stabilization on the first video. I only noticed this because the filesize was much less than the other two videos. Using this incomplete stabilized video, there was no error output during the synthetic bee step, only in the background subtraction. Once I reran the stabilization, that resolved the error.

Sorry for trouble, I should have caught this mistake earlier.

Thank you.
